Cairo Desktop/Cairo Desktop/Desktop.xaml.cs

```diff
@@ -60,6 +61,28 @@ public Desktop()
             setSize();
             setGridPosition();
             setBackground();
+
+            navigationManager = new DynamicDesktopNavigationManager();
+            navigationManager.Navigating += NavigationManager_Navigating;
+
+            Settings.Instance.PropertyChanged += Instance_PropertyChanged;
+        }
+
+        private void Instance_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e != null && !string.IsNullOrWhiteSpace(e.PropertyName))
+            {
+                switch (e.PropertyName)
+                {
+                    case "DesktopBackgroundType":
+                    case "BingWallpaperStyle":
+                    case "CairoBackgroundImagePath":
+                    case "CairoBackgroundImageStyle":
+                    case "CairoBackgroundVideoPath":
+                        ReloadBackground();
```

Comment:
We already handle SPI_SETDESKWALLPAPER, if you broadcast that message when changing the setting then we wouldnt need this right?

Reply:
Actually I think we should only send SPI_SETDESKWALLPAPER only for the Windows background which is addressed in the other comments so I will resolve this one.